app/models/labware_creators/pcr_cycles_binned_plate.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+module LabwareCreators
+  # Handles the generation of a plate with wells binned according to the number of
+  # PCR cycles that has been determined by the customer.
+  # Uploads a file supplied by the customer that has a row per each well and
+  # includes Sample Volume, Diluent Volume, PCR Cycles, Sub-Pool and Coverage columns.
+  # Uses the PCR Cycles column to determine the binning arrangement of the wells,
+  # and the Sample Volume and Diluent Volume columns in the well transfers.
+  # Sub-Pool and Coverage need to be stored for a later step downstream in the pipeline,
+  # at the point where custom pooling is performed.
+  # Wells in the bins are applied to the destination by column order.
+  # If there is enough space on the destination plate each new bin will start in a new
+  # column. Otherwise bins will run consecutively without gaps.
+  #
+  #
+  # Source Plate                  Dest Plate
+  # +--+--+--~                    +--+--+--~
+  # |A1| pcr_cycles = 12 (bin 2)  |B1|A1|C1|
+  # +--+--+--~                    +--+--+--~
+  # |B1| pcr_cycles = 15 (bin 1)  |D1|E1|  |
+  # +--+--+--~  +                 +--+--+--~
+  # |C1| pcr_cycles = 10 (bin 3)  |  |G1|  |
+  # +--+--+--~                    +--+--+--~
+  # |D1| pcr_cycles = 15 (bin 1)  |  |  |  |
+  # +--+--+--~                    +--+--+--~
+  # |E1| pcr_cycles = 12 (bin 2)  |  |  |  |
+  # +--+--+--~                    +--+--+--~
+  # |G1| pcr_cycles = 12 (bin 2)  |  |  |  |
+  class PcrCyclesBinnedPlate < StampedPlate
+    include LabwareCreators::CustomPage
+
+    MISSING_WELL_DETAIL = 'is missing a row for well %s, all wells with content must have a row in the uploaded file.'
+    PENDING_WELL = 'contains at least one pending well %s, the plate and all wells in it should be passed before creating the child plate.'
+
+    self.page = 'pcr_cycles_binned_plate' # TODO: how does this work? new page for uploading file? see custom_pooled_tubes directory in labware_creators
```

Comment:
Do we still need this comment? Seems like it works...

Reply:
Removed